api/envoy/config/route/v3/route.proto

```diff
@@ -113,8 +113,10 @@ message RouteConfiguration {
   //
   // .. warning::
   //
-  //   The content of :ref:`direct response body
-  //   <envoy_api_field_config.route.v3.DirectResponseAction.body>` will be held in memory.
+  //   Envoy currently holds the content of :ref:`direct response body
+  //   <envoy_api_field_config.route.v3.DirectResponseAction.body>` in memory. Be careful setting
+  //   this to be larger than the default 4KB. The default is limited to 4KB to keep the proxy's
```

Comment:
Is this the only reason? My understanding is we also want to avoid flooding watermarked buffers. E.g. if the buffer limits are 128KB and we dump 10MB in a direct response, that will be queued. @alyssawilk for thoughts.

Reply:
Oh, this is a gem. Thanks, @htuch! I didn't know that. I will do more digging. And, yes, I'll be very thankful if @alyssawilk can give some thoughts on this.

Reply:
I don't think it'd be queued, I think we'd disable reading from a nonexistent upstream (and risk OOMs if there were lots of direct responses)
I think it'd be fine to allow larger responses (though I'd be inclined to add a warning note in the API about it not being subject to memory guards so use with care) but suggest an integration test to make sure it all works as I think it should :-)

Reply:
I have added "standard" cases to the direct response integration test. If you think we need to cover more cases, please let me know. re: Buffer limit, I have a test case that sends 1000 * 4K so I think it's enough.